src/components/DocumentEditPage.js

```diff
@@ -0,0 +1,168 @@
+import Editor from "./Editor.js";
+import { request } from "../api/api.js";
+import { push } from "../routes/router.js";
+import { getItem, setItem, removeItem } from "../utils/storage.js";
+
+export default function DocumentEditPage({ $target, initialState }) {
+  const $documentEditPage = document.createElement("div");
+
+  $documentEditPage.className = "documentEditPage";
+
+  $documentEditPage.style.width = "60%";
+  $documentEditPage.style.margin = "auto";
+
+  this.state = initialState;
+
+  let docLocalSaveKey = `temp-document-${this.state.docId}`;
+  let timer = null;
+
+  const editor = new Editor({
+    $target: $documentEditPage,
+    initialState: {
+      title: "Untitled",
+      content: "",
+    },
+    onEditing: (doc) => {
+      setItem(docLocalSaveKey, {
+        ...doc,
+        tempSaveDate: new Date(),
+      });
+
+      if (timer !== null) {
+        clearTimeout(timer);
+      }
+
+      timer = setTimeout(async () => {
+        const isNew = this.state.docId === "new";
+        if (isNew) {
+          if (doc.title === "") {
+            doc.title = "Untitled";
+          }
+
+          const createDocument = await request("/documents", {
+            method: "POST",
+            body: JSON.stringify(doc),
+          });
+
+          await request(`/documents/${createDocument.id}`, {
+            method: "PUT",
+            body: JSON.stringify(doc),
+          });
+
+          history.replaceState(null, null, `/documents/${createDocument.id}`);
+          removeItem(docLocalSaveKey);
+
+          this.setState({
+            docId: createDocument.id,
+          });
+
+          push(createDocument.id);
+        } else {
+          if (doc.title === "") {
+            doc.title = "Untitled";
+          }
+          await request(`/documents/${doc.id}`, {
+            method: "PUT",
+            body: JSON.stringify(doc),
+          });
+
+          removeItem(docLocalSaveKey);
+        }
+      }, 1000);
+    },
+  });
+
+  this.setState = async (nextState) => {
+    if (this.state.docId === "new" && nextState.docId === "new") {
+      const tempDocument = await getItem(docLocalSaveKey, {
+        title: "Untitled",
+        content: "",
+      });
+
+      if (tempDocument.title !== "" || tempDocument.content !== "") {
```

Comment:
```suggestion
      if (!(tempDocument.title === "" && tempDocument.content === "")) {
```

똑같이 동작하는 코드인 것 같아서 제안드려봅니다..! 🤔

Reply:
@jgjgill 제 생각에는 OR연산자가 더 나은 것 같습니다!! AND 연산자는 2개 다 비교해야하지만 OR 연산자 같은 경우 한 가지의 경우만 만족하면 되기 때문입니다 😀

Reply:
소소한 TMI

형변환을 통한 검증도 가능할 것 같아요!

```suggestion
      if (!tempDocument.title || !tempDocument.content) {
```